src/main/authManager.js

```diff
@@ -29,76 +34,31 @@ export class AuthManager {
 
     this.loginCallbackMap.set(request.url, typeof callback === 'undefined' ? null : callback); // if callback is undefined set it to null instead so we know we have set it up with no value
     if (urlUtils.isTrustedURL(request.url, this.config.teams) || urlUtils.isCustomLoginURL(parsedURL, server, this.config.teams) || this.trustedOriginsStore.checkPermission(request.url, BASIC_AUTH_PERMISSION)) {
-      this.addToLoginQueue(request, authInfo);
+      this.popLoginModal(request, authInfo);
     } else {
-      this.addToPermissionQueue(request, authInfo, BASIC_AUTH_PERMISSION);
+      this.popPermissionModal(request, authInfo, BASIC_AUTH_PERMISSION);
     }
   }
 
-  addToPermissionQueue = (request, authInfo, permission) => {
-    this.permissionQueue.push({
-      request,
-      authInfo,
-      permission,
+  popLoginModal = (request, authInfo) => {
+    const modalPromise = addModal(`login-${request.url}`, loginModalHtml, modalPreload, {request, authInfo}, WindowManager.getMainWindow());
+    modalPromise.then((data) => {
+      const {username, password} = data;
+      this.handleLoginCredentialsEvent(request, username, password);
+    }).catch(() => {
```

Comment:
since catch is for user cancel and errors, could we throw the info to the console to ease future debugging?